rasa/train.py

```diff
@@ -491,19 +491,27 @@ async def _train_core_with_validated_data(
         return _train_path
 
 
-def _core_model_for_finetuning(
+async def _core_model_for_finetuning(
     model_to_finetune: Text,
-    new_config: Optional[Dict] = None,
+    file_importer: TrainingDataImporter,
     finetuning_epoch_fraction: float = 1.0,
 ) -> Optional[Agent]:
     path_to_archive = model.get_model_for_finetuning(model_to_finetune)
     if not path_to_archive:
         return None
 
     with model.unpack_model(path_to_archive) as unpacked:
+        new_fingerprint = await model.model_fingerprint(file_importer)
+        old_fingerprint = model.fingerprint_from_path(unpacked)
+        if not model.can_finetune(old_fingerprint, new_fingerprint, core=True):
```

Comment:
technically we only need to do this when `train_path is None`, correct?

Reply:
Technically yes - although maybe it would be a bit confusing? Also maybe a bit brittle as it relies on the fact we check NLU first? I'm not sure WDYT?

Reply:
Both things are fine for a first iteration in my opinion

Reply:
Oops sorry forgot this was still outstanding.

Reply:
no worries. If it'd been more important, I would have made it clearer